frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -413,18 +418,21 @@ function DatasetEditor(props) {
     [datasetEditorTab, renderSelectableTableColumnHeader],
   );
 
-  const canSaveChanges = useMemo(() => {
-    const { isNative } = Lib.queryDisplayInfo(dataset.query());
-    const isEmpty = !isNative
-      ? Lib.databaseID(dataset.query()) == null
-      : dataset.legacyQuery().isEmpty();
+  const { isNative } = Lib.queryDisplayInfo(dataset.query());
+  const isEmpty = !isNative
+    ? Lib.databaseID(dataset.query()) == null
+    : dataset.legacyQuery().isEmpty();
 
-    if (isEmpty) {
-      return false;
-    }
-    const everyFieldHasDisplayName = fields.every(field => field.display_name);
-    return everyFieldHasDisplayName && isDirty;
-  }, [dataset, fields, isDirty]);
+  const canSaveChanges =
+    !isEmpty &&
+    isDirty &&
+    (!isNative || !isResultDirty) &&
+    fields.every(field => field.display_name);
+
+  const saveButtonTooltipLabel =
+    !isEmpty && isDirty && isNative && isResultDirty
```

Comment:
Why only native models?

Reply:
Because for MBQL queries `result_metadata` can be computed by the BE without running the query